ios/packages/core/Sources/utilities/JSValue+Extensions.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  JSValue+Extensions.swift
+//  PlayerUI
+//
+//  Created by Zhao Xia Wu on 2024-01-18.
+//
+
+import Foundation
+import JavaScriptCore
+
+extension JSValue {
+    var tryCatchWrapper: JSValue? {
+        self.context.evaluateScript(
+        """
+           (fn, args) => {
+               try {
+                   console.log(args)
+                   return fn(...args)
+               } catch(e) {
+                   console.log(e)
+                   if (e instanceof Error) {
+
+                      return e
+                }
+             }
+           }
+        """)
+
+    }
+
+    var errorCheckWrapper: JSValue? {
+        self.context.evaluateScript(
+        """
+                (obj) => (obj instanceof Error)
+        """)
+    }
+
+
+    /**
+     A way to catch errors for void functions not called inside a player process. Called on void functions.
+     - parameters:
+        - args: List of arguments taken by the function
+     */
+    public func callTryCatchWrapper(args: Any...) throws {
+        let result = self.tryCatchWrapper?.call(withArguments: [self, args])
+
+        let isError = self.errorCheckWrapper?.call(withArguments: [result as Any])
+
+        if isError?.toBool() == true {
+            throw JSValueError.thrownFromJS
+        }
+    }
+
+
+    /**
+     A way to catch errors for functions not called inside a player process. Called on functions with a return value.
+     - parameters:
+        - args: List of arguments taken by the function
+     */
+    public func callTryCatchWrapperWithReturnValue(args: Any...) throws -> JSValue? {
```

Comment:
you could potentially just use this one and mark it `@discardableResult` since a `Void` function in JS would still return `undefined` technically